crate_universe/src/renderer.rs

```diff
@@ -485,3 +526,285 @@ enum Kind {
     Build,
     ProcMacro,
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::testing;
+
+    use indoc::indoc;
+    use maplit::{btreemap, btreeset};
+
+    fn mock_renderer(git: bool) -> Renderer {
+        Renderer::new(
+            RenderConfig {
+                repo_rule_name: String::from("rule_prefix"),
+                crate_registry_template: String::from(
+                    "https://crates.io/api/v1/crates/{crate}/{version}/download",
+                ),
+                rules_rust_workspace_name: String::from("rules_rust"),
+            },
+            String::from("598"),
+            vec![testing::lazy_static_crate_context(git)],
+            Dependencies {
+                normal: BTreeMap::new(),
+                build: BTreeMap::new(),
+                dev: BTreeMap::new(),
+            },
+            BTreeMap::new(),
+        )
+    }
+
+    #[test]
+    fn render_http_archive() {
+        let renderer = mock_renderer(false);
+
+        let mut output = Vec::new();
+
+        renderer
+            .render_workspaces(&mut output)
+            .expect("Error rendering");
+
+        let output = String::from_utf8(output).expect("Non-UTF8 output");
+
+        // TODO: Don't unconditionally load new_git_repository and http_archive
+        let expected_repository_rule = indoc! { r#"
+            # rules_rust crate_universe file format 1
+            # config hash 598
+
+            load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")
+            load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+            def pinned_rust_install():
+                http_archive(
+                    name = "rule_prefix__lazy_static__1_4_0",
+                    build_file = Label("//:BUILD.lazy_static-1.4.0.bazel"),
+                    sha256 = "e2abad23fbc42b3700f2f279844dc832adb2b2eb069b2df918f455c4e18cc646",
+                    strip_prefix = "lazy_static-1.4.0",
+                    type = "tar.gz",
+                    url = "https://crates.io/api/v1/crates/lazy_static/1.4.0/download",
+                )
+
+            "# };
+
+        assert_eq!(output, expected_repository_rule);
+    }
+
+    #[test]
+    fn render_git_repository() {
+        let renderer = mock_renderer(true);
+
+        let mut output = Vec::new();
+
+        renderer
+            .render_workspaces(&mut output)
+            .expect("Error rendering");
+
+        let output = String::from_utf8(output).expect("Non-UTF8 output");
+
+        // TODO: Don't unconditionally load new_git_repository and http_archive
+        let expected_repository_rule = indoc! { r#"
+            # rules_rust crate_universe file format 1
+            # config hash 598
+            
```

Comment:
We should drop these lines from the output, they're only relevant for the lockfile which no longer uses them.